### PR TITLE
[gestures] Introduce pinchScrollEnabled config.

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

This PR introduces `pinchScrollEnabled` config to gestures plugin, when enabled, the 2-finger scroll gesture is enabled, default to true.

Note this is a behaviour change compared with previous release, however the current default behaviour matches Mapbox Maps iOS SDK and it unblocks the Free Drag gesture with 2 touches, meaning:

> Adjusts the center, zoom, and rotate simultaneously. Continuous. Exclusive. The coordinates under the touches when the gesture began always remains under the touches. Adjustment ends immediately when the gesture ends

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Mapbox welcomes participation and contributions from everyone.
 * Use a single shared buffer across all globe tiles to increase globe rendering performance. ([#1351](https://github.com/mapbox/mapbox-maps-android/pull/1351))
 * Re-introduce partial tile loading feature that decreases map load times. ([#1351](https://github.com/mapbox/mapbox-maps-android/pull/1351))
 * The `TilesetDescriptorOptions.pixelRatio` parameter is now passed to the TileStore and considered for the raster tile pack loading. This enables loading of a raster tilepacks for retina displays. ([#1351](https://github.com/mapbox/mapbox-maps-android/pull/1351))
+* Introduce pinchScrollEnabled configuration, default to true. ([#1343](https://github.com/mapbox/mapbox-maps-android/pull/1343))
```
Comment (kiryldz) [inline CHANGELOG.md:14]: perhaps drop a note what it means like `enabling 2-finger map panning`?

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 2]
```diff
@@ -1378,6 +1378,11 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
         return false
       }
 
+      // Skip 2-finger scroll if pinchScrollEnabled is disabled.
+      if (!internalSettings.pinchScrollEnabled && detector.pointersCount > 1) {
```
Comment (kiryldz) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1382]: shouldn't we actually check for `detector.pointersCount > 1` instead?

Comment (pengdev) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1382]: @kiryldz we are checking `detector.pointersCount > 1` and `pinchScrollEnabled` here, or did I miss anything?

File: CHANGELOG.md (hunk 1 above)
Comment (pengdev) [reply CHANGELOG.md:14]:
```suggestion
* Introduce pinchScrollEnabled configuration for enabling 2-finger map panning, default to true. ([#1343](https://github.com/mapbox/mapbox-maps-android/pull/1343))
```

File: plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt [hunk 3]
```diff
@@ -423,6 +423,51 @@ class GesturesPluginTest {
     verify { cameraAnimationsPlugin.easeTo(any(), any()) }
   }
 
+  @Test
+  fun verifyMoveListenerPinchScrollDisabled() {
```
Comment (Chaoba) [inline plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt:427]: Would be nice if add a verifyMoveListenerPinchScrollEnabled

Comment (pengdev) [reply plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt:427]: The enabled case is the default behaviour and is covered in the above test case. Or do you mean specifically call the `presenter.pinchScrollEnabled = true`?

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt (hunk 2 above)
Comment (kiryldz) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1382]: oops, sorry, my bad, nevermind 🤦

**Review by kiryldz — APPROVED** [review]

File: plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt (hunk 3 above)
Comment (pengdev) [reply plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt:427]: Added a separate test case for setting it to true.

**Review by Chaoba — APPROVED** [review]

**Pull request merged**